SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -235,5 +241,69 @@ internal class InAppReceipt {
             }
             task.resume()
     }
-    
+
+    /**
+     *  Verify the purchase of a product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter inReceipt: the receipt to test in
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     */
+    class func verifyPurchase(
+      productId productId: String,
+      inReceipt receipt: ReceiptInfo,
+      validUntil: NSDate? = nil
+    ) -> SwiftyStoreKit.VerifyPurchaseResult {
+      
+          // Get all receipts
+          guard let allReceipts = receipt["receipt"]?["in_app"] as? [ReceiptInfo] else {
+              return .NotPurchased
+          }
+      
+          // Filter receipts with matching product id
+          let receiptsMatchingProductId = allReceipts
+          .filter { (receipt) -> Bool in
+                let product_id = receipt["product_id"] as? String
+                return product_id == productId
+          }
+      
+          // Verify that at least one receipt has the right product id
+          guard receiptsMatchingProductId.count >= 1 else {
+              return .NotPurchased
+          }
+      
+          guard let validUntil = validUntil else {
+              // Do not need to verify the expiration date
+              return .Purchased(expiresDate: nil)
+          }
+      
+          // Return the expires dates sorted desc
+          let expiresDates = receiptsMatchingProductId
+          .map { (receipt) -> NSDate in
+            // TODO: should use 'expires_date' or 'expiration_date' instead? See:
+            // https://developer.apple.com/library/mac/releasenotes/General/ValidateAppStoreReceipt/Chapters/ReceiptFields.html
+              let expires_date = receipt["expires_date_ms"] as? NSString
+              let expires_date_double = (expires_date?.doubleValue ?? 0.0) / 1000
+              return NSDate(timeIntervalSince1970: expires_date_double)
```

Comment:
I'm not sure this code works as there isn't an `expires_date_ms` field [here](https://developer.apple.com/library/mac/releasenotes/General/ValidateAppStoreReceipt/Chapters/ReceiptFields.html).
In my tests I have added a consumable IAP and I have observed that after purchasing it successfully and calling the verifyPurchase() method, this call will return `NSDate(timeIntervalSince1970: 0)` which is _1 January 1970_ and means that the receipt is always expired.
By reading the docs it seems that if the `expires_date` or `expiration_date` fields are missing, the IAP can't expire and we should return `.Purchased(expiresDate: nil)`.
Instead, if we have an expiration date, we should compare it with the current date, and return `Purchased` or `Expired` based on that.
What do you think?